### DISCUSSION TITLE
Error installing tsfresh with pip - matrixprofile

### DISCUSSION THREAD

**Discussion by tccf1109** [blue-yonder/tsfresh#927] [Q&A]
Hello,

I was trying to use tsfresh and began by installing it with pip install tsfresh command. During the command execution I got an error installing matrixprofile. I really would love to use tsfresh in my app, and deploy it in a Heroku server. However I can't solve this problem.

pip install tsfresh           
Collecting tsfresh
  Using cached tsfresh-0.19.0-py2.py3-none-any.whl (97 kB)
Requirement already satisfied: dask[dataframe]>=2.9.0 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from tsfresh) (2022.1.1)
Requirement already satisfied: scikit-learn>=0.22.0 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from tsfresh) (1.0.2)
Requirement already satisfied: numpy>=1.15.1 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from tsfresh) (1.21.5)
Requirement already satisfied: tqdm>=4.10.0 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from tsfresh) (4.62.3)
Collecting matrixprofile<2.0.0,>=1.1.10
  Using cached matrixprofile-1.1.10.tar.gz (331 kB)
  Preparing metadata (setup.py) ... done
Requirement already satisfied: requests>=2.9.1 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from tsfresh) (2.27.1)
Requirement already satisfied: statsmodels>=0.13 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from tsfresh) (0.13.2)
Requirement already satisfied: pandas>=0.25.0 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from tsfresh) (1.3.5)
Requirement already satisfied: cloudpickle in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from tsfresh) (2.0.0)
Requirement already satisfied: patsy>=0.4.1 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from tsfresh) (0.5.2)
Requirement already satisfied: scipy>=1.2.0 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from tsfresh) (1.7.3)
Collecting distributed>=2.11.0
  Using cached distributed-2022.1.1-py3-none-any.whl (830 kB)
Requirement already satisfied: stumpy>=1.7.2 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from tsfresh) (1.10.2)
Requirement already satisfied: toolz>=0.8.2 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from dask[dataframe]>=2.9.0->tsfresh) (0.11.2)
Requirement already satisfied: partd>=0.3.10 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from dask[dataframe]>=2.9.0->tsfresh) (1.2.0)
Requirement already satisfied: fsspec>=0.6.0 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from dask[dataframe]>=2.9.0->tsfresh) (2022.1.0)
Requirement already satisfied: packaging>=20.0 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from dask[dataframe]>=2.9.0->tsfresh) (21.3)
Requirement already satisfied: pyyaml>=5.3.1 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from dask[dataframe]>=2.9.0->tsfresh) (6.0)
Requirement already satisfied: click>=6.6 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from distributed>=2.11.0->tsfresh) (7.1.2)
Requirement already satisfied: setuptools in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from distributed>=2.11.0->tsfresh) (58.1.0)
Requirement already satisfied: tblib>=1.6.0 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from distributed>=2.11.0->tsfresh) (1.7.0)
Requirement already satisfied: jinja2 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from distributed>=2.11.0->tsfresh) (3.0.3)
Requirement already satisfied: tornado>=6.0.3 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from distributed>=2.11.0->tsfresh) (6.1)
Requirement already satisfied: msgpack>=0.6.0 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from distributed>=2.11.0->tsfresh) (1.0.3)
Requirement already satisfied: zict>=0.1.3 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from distributed>=2.11.0->tsfresh) (2.0.0)
Requirement already satisfied: sortedcontainers!=2.0.0,!=2.0.1 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from distributed>=2.11.0->tsfresh) (2.4.0)
Requirement already satisfied: psutil>=5.0 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from distributed>=2.11.0->tsfresh) (5.9.0)
Requirement already satisfied: matplotlib>=3.0.3 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from matrixprofile<2.0.0,>=1.1.10->tsfresh) (3.5.1)
Collecting protobuf==3.11.2
  Using cached protobuf-3.11.2-py2.py3-none-any.whl (434 kB)
Requirement already satisfied: six>=1.9 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from protobuf==3.11.2->matrixprofile<2.0.0,>=1.1.10->tsfresh) (1.16.0)
Requirement already satisfied: pytz>=2017.3 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from pandas>=0.25.0->tsfresh) (2021.3)
Requirement already satisfied: python-dateutil>=2.7.3 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from pandas>=0.25.0->tsfresh) (2.8.2)
Requirement already satisfied: charset-normalizer~=2.0.0 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from requests>=2.9.1->tsfresh) (2.0.10)
Requirement already satisfied: idna<4,>=2.5 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from requests>=2.9.1->tsfresh) (3.3)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from requests>=2.9.1->tsfresh) (1.26.8)
Requirement already satisfied: certifi>=2017.4.17 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from requests>=2.9.1->tsfresh) (2021.10.8)
Requirement already satisfied: joblib>=0.11 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from scikit-learn>=0.22.0->tsfresh) (1.1.0)
Requirement already satisfied: threadpoolctl>=2.0.0 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from scikit-learn>=0.22.0->tsfresh) (3.0.0)
Requirement already satisfied: numba>=0.48 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from stumpy>=1.7.2->tsfresh) (0.55.1)
Requirement already satisfied: colorama in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from tqdm>=4.10.0->tsfresh) (0.4.4)
Requirement already satisfied: pillow>=6.2.0 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from matplotlib>=3.0.3->matrixprofile<2.0.0,>=1.1.10->tsfresh) (9.0.0)
Requirement already satisfied: kiwisolver>=1.0.1 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from matplotlib>=3.0.3->matrixprofile<2.0.0,>=1.1.10->tsfresh) (1.3.2)
Requirement already satisfied: pyparsing>=2.2.1 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from matplotlib>=3.0.3->matrixprofile<2.0.0,>=1.1.10->tsfresh) (3.0.6)
Requirement already satisfied: cycler>=0.10 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from matplotlib>=3.0.3->matrixprofile<2.0.0,>=1.1.10->tsfresh) (0.11.0)
Requirement already satisfied: fonttools>=4.22.0 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from matplotlib>=3.0.3->matrixprofile<2.0.0,>=1.1.10->tsfresh) (4.28.5)
Requirement already satisfied: llvmlite<0.39,>=0.38.0rc1 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from numba>=0.48->stumpy>=1.7.2->tsfresh) (0.38.0)
Requirement already satisfied: locket in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from partd>=0.3.10->dask[dataframe]>=2.9.0->tsfresh) (0.2.1)
Requirement already satisfied: heapdict in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from zict>=0.1.3->distributed>=2.11.0->tsfresh) (1.0.1)
Requirement already satisfied: MarkupSafe>=2.0 in c:\users\35191\appdata\local\programs\python\python310\lib\site-packages (from jinja2->distributed>=2.11.0->tsfresh) (2.0.1)
Building wheels for collected packages: matrixprofile
  Building wheel for matrixprofile (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [97 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-3.10
      creating build\lib.win-amd64-3.10\matrixprofile
      copying matrixprofile\analyze.py -> build\lib.win-amd64-3.10\matrixprofile
      copying matrixprofile\compute.py -> build\lib.win-amd64-3.10\matrixprofile
      copying matrixprofile\core.py -> build\lib.win-amd64-3.10\matrixprofile
      copying matrixprofile\discover.py -> build\lib.win-amd64-3.10\matrixprofile
      copying matrixprofile\exceptions.py -> build\lib.win-amd64-3.10\matrixprofile
      copying matrixprofile\preprocess.py -> build\lib.win-amd64-3.10\matrixprofile
      copying matrixprofile\transform.py -> build\lib.win-amd64-3.10\matrixprofile
      copying matrixprofile\utils.py -> build\lib.win-amd64-3.10\matrixprofile
      copying matrixprofile\version.py -> build\lib.win-amd64-3.10\matrixprofile
      copying matrixprofile\visualize.py -> build\lib.win-amd64-3.10\matrixprofile
      copying matrixprofile\__init__.py -> build\lib.win-amd64-3.10\matrixprofile
      creating build\lib.win-amd64-3.10\tests
      copying tests\test_analyze.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_annotation_vector.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_compute.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_core.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_cycore.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_datasets.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_hierarchical_clustering.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_io.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_mass2.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_mpdist.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_mpx.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_pairwise_dist.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_preprocess.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_regimes.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_scrimp.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_skimp.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_snippets.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_statistics.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_stomp.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_top_k_discords.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_top_k_motifs.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_utils.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_visualize.py -> build\lib.win-amd64-3.10\tests
      copying tests\__init__.py -> build\lib.win-amd64-3.10\tests
      creating build\lib.win-amd64-3.10\matrixprofile\algorithms
      copying matrixprofile\algorithms\hierarchical_clustering.py -> build\lib.win-amd64-3.10\matrixprofile\algorithms   
      copying matrixprofile\algorithms\mass2.py -> build\lib.win-amd64-3.10\matrixprofile\algorithms
      copying matrixprofile\algorithms\mpdist.py -> build\lib.win-amd64-3.10\matrixprofile\algorithms
      copying matrixprofile\algorithms\mpx.py -> build\lib.win-amd64-3.10\matrixprofile\algorithms
      copying matrixprofile\algorithms\pairwise_dist.py -> build\lib.win-amd64-3.10\matrixprofile\algorithms
      copying matrixprofile\algorithms\regimes.py -> build\lib.win-amd64-3.10\matrixprofile\algorithms
      copying matrixprofile\algorithms\scrimp.py -> build\lib.win-amd64-3.10\matrixprofile\algorithms
      copying matrixprofile\algorithms\skimp.py -> build\lib.win-amd64-3.10\matrixprofile\algorithms
      copying matrixprofile\algorithms\snippets.py -> build\lib.win-amd64-3.10\matrixprofile\algorithms
      copying matrixprofile\algorithms\statistics.py -> build\lib.win-amd64-3.10\matrixprofile\algorithms
      copying matrixprofile\algorithms\stomp.py -> build\lib.win-amd64-3.10\matrixprofile\algorithms
      copying matrixprofile\algorithms\top_k_discords.py -> build\lib.win-amd64-3.10\matrixprofile\algorithms
      copying matrixprofile\algorithms\top_k_motifs.py -> build\lib.win-amd64-3.10\matrixprofile\algorithms
      copying matrixprofile\algorithms\__init__.py -> build\lib.win-amd64-3.10\matrixprofile\algorithms
      creating build\lib.win-amd64-3.10\matrixprofile\datasets
      copying matrixprofile\datasets\datasets.py -> build\lib.win-amd64-3.10\matrixprofile\datasets
      copying matrixprofile\datasets\__init__.py -> build\lib.win-amd64-3.10\matrixprofile\datasets
      creating build\lib.win-amd64-3.10\matrixprofile\io
      copying matrixprofile\io\__init__.py -> build\lib.win-amd64-3.10\matrixprofile\io
      copying matrixprofile\io\__io.py -> build\lib.win-amd64-3.10\matrixprofile\io
      creating build\lib.win-amd64-3.10\matrixprofile\io\protobuf
      copying matrixprofile\io\protobuf\protobuf_utils.py -> build\lib.win-amd64-3.10\matrixprofile\io\protobuf
      copying matrixprofile\io\protobuf\proto_messages_pb2.py -> build\lib.win-amd64-3.10\matrixprofile\io\protobuf      
      copying matrixprofile\io\protobuf\__init__.py -> build\lib.win-amd64-3.10\matrixprofile\io\protobuf
      running egg_info
      writing matrixprofile.egg-info\PKG-INFO
      writing dependency_links to matrixprofile.egg-info\dependency_links.txt
      writing requirements to matrixprofile.egg-info\requires.txt
      writing top-level names to matrixprofile.egg-info\top_level.txt
      reading manifest file 'matrixprofile.egg-info\SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      warning: no files found matching '*.csv' under directory '.'
      warning: no files found matching '*.json' under directory '.'
      adding license file 'LICENSE'
      writing manifest file 'matrixprofile.egg-info\SOURCES.txt'
      copying matrixprofile\cycore.c -> build\lib.win-amd64-3.10\matrixprofile
      copying matrixprofile\cycore.pyx -> build\lib.win-amd64-3.10\matrixprofile
      copying matrixprofile\algorithms\cympx.c -> build\lib.win-amd64-3.10\matrixprofile\algorithms
      copying tests\mpx_ab_mpa.txt -> build\lib.win-amd64-3.10\tests
      copying tests\mpx_ab_mpb.txt -> build\lib.win-amd64-3.10\tests
      copying tests\mpx_ab_mpia.txt -> build\lib.win-amd64-3.10\tests
      copying tests\mpx_ab_mpib.txt -> build\lib.win-amd64-3.10\tests
      copying tests\mpx_mp.txt -> build\lib.win-amd64-3.10\tests
      copying tests\mpx_mpi.txt -> build\lib.win-amd64-3.10\tests
      copying tests\muinvn_mua.txt -> build\lib.win-amd64-3.10\tests
      copying tests\muinvn_stda.txt -> build\lib.win-amd64-3.10\tests
      copying tests\sampledata.txt -> build\lib.win-amd64-3.10\tests
      copying tests\scrimp.mp.txt -> build\lib.win-amd64-3.10\tests
      copying tests\scrimp.mpi.txt -> build\lib.win-amd64-3.10\tests
      copying matrixprofile\algorithms\cympx.pyx -> build\lib.win-amd64-3.10\matrixprofile\algorithms
      copying matrixprofile\io\protobuf\proto_messages.proto -> build\lib.win-amd64-3.10\matrixprofile\io\protobuf       
      running build_ext
      building 'matrixprofile.algorithms.cympx' extension
      error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for matrixprofile
  Running setup.py clean for matrixprofile
Failed to build matrixprofile
Installing collected packages: protobuf, matrixprofile, distributed, tsfresh
  Attempting uninstall: protobuf
    Found existing installation: protobuf 3.19.4
    Uninstalling protobuf-3.19.4:
      Successfully uninstalled protobuf-3.19.4
  Running setup.py install for matrixprofile ... error
  error: subprocess-exited-with-error

  × Running setup.py install for matrixprofile did not run successfully.
  │ exit code: 1
  ╰─> [97 lines of output]
      running install
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-3.10
      creating build\lib.win-amd64-3.10\matrixprofile
      copying matrixprofile\analyze.py -> build\lib.win-amd64-3.10\matrixprofile
      copying matrixprofile\compute.py -> build\lib.win-amd64-3.10\matrixprofile
      copying matrixprofile\core.py -> build\lib.win-amd64-3.10\matrixprofile
      copying matrixprofile\discover.py -> build\lib.win-amd64-3.10\matrixprofile
      copying matrixprofile\exceptions.py -> build\lib.win-amd64-3.10\matrixprofile
      copying matrixprofile\preprocess.py -> build\lib.win-amd64-3.10\matrixprofile
      copying matrixprofile\transform.py -> build\lib.win-amd64-3.10\matrixprofile
      copying matrixprofile\utils.py -> build\lib.win-amd64-3.10\matrixprofile
      copying matrixprofile\version.py -> build\lib.win-amd64-3.10\matrixprofile
      copying matrixprofile\visualize.py -> build\lib.win-amd64-3.10\matrixprofile
      copying matrixprofile\__init__.py -> build\lib.win-amd64-3.10\matrixprofile
      creating build\lib.win-amd64-3.10\tests
      copying tests\test_analyze.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_annotation_vector.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_compute.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_core.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_cycore.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_datasets.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_hierarchical_clustering.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_io.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_mass2.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_mpdist.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_mpx.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_pairwise_dist.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_preprocess.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_regimes.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_scrimp.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_skimp.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_snippets.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_statistics.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_stomp.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_top_k_discords.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_top_k_motifs.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_utils.py -> build\lib.win-amd64-3.10\tests
      copying tests\test_visualize.py -> build\lib.win-amd64-3.10\tests
      copying tests\__init__.py -> build\lib.win-amd64-3.10\tests
      creating build\lib.win-amd64-3.10\matrixprofile\algorithms
      copying matrixprofile\algorithms\hierarchical_clustering.py -> build\lib.win-amd64-3.10\matrixprofile\algorithms   
      copying matrixprofile\algorithms\mass2.py -> build\lib.win-amd64-3.10\matrixprofile\algorithms
      copying matrixprofile\algorithms\mpdist.py -> build\lib.win-amd64-3.10\matrixprofile\algorithms
      copying matrixprofile\algorithms\mpx.py -> build\lib.win-amd64-3.10\matrixprofile\algorithms
      copying matrixprofile\algorithms\pairwise_dist.py -> build\lib.win-amd64-3.10\matrixprofile\algorithms
      copying matrixprofile\algorithms\regimes.py -> build\lib.win-amd64-3.10\matrixprofile\algorithms
      copying matrixprofile\algorithms\scrimp.py -> build\lib.win-amd64-3.10\matrixprofile\algorithms
      copying matrixprofile\algorithms\skimp.py -> build\lib.win-amd64-3.10\matrixprofile\algorithms
      copying matrixprofile\algorithms\snippets.py -> build\lib.win-amd64-3.10\matrixprofile\algorithms
      copying matrixprofile\algorithms\statistics.py -> build\lib.win-amd64-3.10\matrixprofile\algorithms
      copying matrixprofile\algorithms\stomp.py -> build\lib.win-amd64-3.10\matrixprofile\algorithms
      copying matrixprofile\algorithms\top_k_discords.py -> build\lib.win-amd64-3.10\matrixprofile\algorithms
      copying matrixprofile\algorithms\top_k_motifs.py -> build\lib.win-amd64-3.10\matrixprofile\algorithms
      copying matrixprofile\algorithms\__init__.py -> build\lib.win-amd64-3.10\matrixprofile\algorithms
      creating build\lib.win-amd64-3.10\matrixprofile\datasets
      copying matrixprofile\datasets\datasets.py -> build\lib.win-amd64-3.10\matrixprofile\datasets
      copying matrixprofile\datasets\__init__.py -> build\lib.win-amd64-3.10\matrixprofile\datasets
      creating build\lib.win-amd64-3.10\matrixprofile\io
      copying matrixprofile\io\__init__.py -> build\lib.win-amd64-3.10\matrixprofile\io
      copying matrixprofile\io\__io.py -> build\lib.win-amd64-3.10\matrixprofile\io
      creating build\lib.win-amd64-3.10\matrixprofile\io\protobuf
      copying matrixprofile\io\protobuf\protobuf_utils.py -> build\lib.win-amd64-3.10\matrixprofile\io\protobuf
      copying matrixprofile\io\protobuf\proto_messages_pb2.py -> build\lib.win-amd64-3.10\matrixprofile\io\protobuf      
      copying matrixprofile\io\protobuf\__init__.py -> build\lib.win-amd64-3.10\matrixprofile\io\protobuf
      running egg_info
      writing matrixprofile.egg-info\PKG-INFO
      writing dependency_links to matrixprofile.egg-info\dependency_links.txt
      writing requirements to matrixprofile.egg-info\requires.txt
      writing top-level names to matrixprofile.egg-info\top_level.txt
      reading manifest file 'matrixprofile.egg-info\SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      warning: no files found matching '*.csv' under directory '.'
      warning: no files found matching '*.json' under directory '.'
      adding license file 'LICENSE'
      writing manifest file 'matrixprofile.egg-info\SOURCES.txt'
      copying matrixprofile\cycore.c -> build\lib.win-amd64-3.10\matrixprofile
      copying matrixprofile\cycore.pyx -> build\lib.win-amd64-3.10\matrixprofile
      copying matrixprofile\algorithms\cympx.c -> build\lib.win-amd64-3.10\matrixprofile\algorithms
      copying tests\mpx_ab_mpa.txt -> build\lib.win-amd64-3.10\tests
      copying tests\mpx_ab_mpb.txt -> build\lib.win-amd64-3.10\tests
      copying tests\mpx_ab_mpia.txt -> build\lib.win-amd64-3.10\tests
      copying tests\mpx_ab_mpib.txt -> build\lib.win-amd64-3.10\tests
      copying tests\mpx_mp.txt -> build\lib.win-amd64-3.10\tests
      copying tests\mpx_mpi.txt -> build\lib.win-amd64-3.10\tests
      copying tests\muinvn_mua.txt -> build\lib.win-amd64-3.10\tests
      copying tests\muinvn_stda.txt -> build\lib.win-amd64-3.10\tests
      copying tests\sampledata.txt -> build\lib.win-amd64-3.10\tests
      copying tests\scrimp.mp.txt -> build\lib.win-amd64-3.10\tests
      copying tests\scrimp.mpi.txt -> build\lib.win-amd64-3.10\tests
      copying matrixprofile\algorithms\cympx.pyx -> build\lib.win-amd64-3.10\matrixprofile\algorithms
      copying matrixprofile\io\protobuf\proto_messages.proto -> build\lib.win-amd64-3.10\matrixprofile\io\protobuf       
      running build_ext
      building 'matrixprofile.algorithms.cympx' extension
      error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> matrixprofile

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.

**Comment by kempa-liehr** [comment]
Hi, 
this installation problem seems to be specific to the matrixprofile package. 
I suggest that you contact the developers of matrixprofile. 
Sorry, for not being more helpful. 
Cheers,Andreas
